Doc/library/shutil.rst

```diff
@@ -363,9 +363,11 @@ Directory and files operations
    Recursively move a file or directory (*src*) to another location (*dst*)
    and return the destination.
 
-   If the destination is an existing directory, then *src* is moved inside that
-   directory. If the destination already exists but is not a directory, it may
-   be overwritten depending on :func:`os.rename` semantics.
+   If the destination is a directory or a symlink to a directory, the source
+   is moved inside the directory. The destination path must not already exist.
```

Comment:
It is a contradiction. "If the destination is a directory or a symlink to a directory ...", but "The destination path must not already exist." If it doesn't exist, it can be a directory or a symlink.

Reply:
Yes, it is confusing to me, too.

According to the code, the "destination" in the second sentence should be referring to the new path of the `src` file/directory after it has been moved into the `dst` directory. (`real_dst`)
https://github.com/python/cpython/blob/5ecfd750b4f511f270c38f0d748da9cffa279295/Lib/shutil.py#L896

Would "The new destination path in that directory must not already exist." be clearer? or do you have better ideas?

Reply:
Maybe something like "If *dst* is an existing directory or  a symbolic link to a directory, ..."? The paragraph above ("to another location (*dst*)") should also be corrected, because another location is not always *dst*. Use "*dst*" to refer to the parameter and "destination" for real destination (returned by this function), based on *dst*, but not always *dst*.